emission/net/ext_service/geocoder/nominatim.py

```diff
@@ -94,6 +99,82 @@ def reverse_geocode(cls, lat, lng):
             return _do_google_reverse(lat, lng) # Just in case
 
 ## Failsafe section
+def check_against_business_location(location='0, 0', address = ''):
+    if not re.compile('^(\-?\d+(\.\d+)?),\s*(\-?\d+(\.\d+)?)$').match(location):
+        raise ValueError('Location Invalid')
+    base_url = NEARBY_URL
+    location = 'location=' + location
+    try:
+        key_string = '&key=' + GOOGLE_MAPS_KEY
+        radius = '&radius=10'
+        url = base_url + location + radius + key_string
+        result = requests.get(url).json()
+        cleaned = result['results']
+        for i in cleaned:
+            #If the street address matches the street address of this business, we return a tuple
+            #signifying success and the business name
+            if address == i['vicinity']:
+                return (True, i['name'])
+        else:
+            return (False, '')
+    except:
+        try:
+            key_string = '&key=' + BACKUP_GOOGLE_MAPS_KEY
+            radius = '&radius=10'
+            url = base_url + location + radius + key_string
+            result = requests.get(url).json()
+            cleaned = result['results']
+            for i in cleaned:
+                if address == i['vicinity']:
+                    return (True, i['name'])
+            else:
+                return (False, '')
+        except:
+            raise ValueError("Something went wrong")
+
+def return_address_from_location_google(location='0,0'):
```

Comment:
ditto

Reply:
isn't there an existing reverse geocode implementation for google in here? why not just re-use that?